### PR TITLE
fix/reduce x-axis label congestion in single view charts

### PR DESCRIPTION
**Pull request by kmeraz**
image
image

### CODE REVIEW

File: .gitignore
Comment (kmeraz) [inline .gitignore:33]: sneaking this in here

File: sections/exchange/TradeCard/Charts/Types/AreaChart.tsx
Comment (kmeraz) [inline sections/exchange/TradeCard/Charts/Types/AreaChart.tsx:85]: two ternary operators based on same predicate due to Recharts not rendering <Fragment/> elements

File: .gitignore
Comment (JChiaramonte7) [reply .gitignore:33]: is this for custom workspace configs?

File: sections/exchange/TradeCard/Charts/Types/AreaChart.tsx
Comment (JChiaramonte7) [reply sections/exchange/TradeCard/Charts/Types/AreaChart.tsx:85]: this is because you had to render day and month axis separately right? 

what happens if there are two months in the x-axis?

File: sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx
Comment (JChiaramonte7) [inline sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx:28]: Somewhere in here it thinks the candlestick month is Jan

image

**Review by JChiaramonte7 — CHANGES_REQUESTED** [review]
Also not related to your PR, but I didn't realize our Month area chart data is reversed? I'm curious where that is happening.

File: .gitignore
Comment (kmeraz) [reply .gitignore:33]: yup yup

File: sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx
Comment (kmeraz) [reply sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx:28]: so it looks like the data coming back from the subgraphs isn't entirelyyyy uniform. For the:
- `useCombinedRates` hook, the timestamps come back as numbers in milliseconds
- `useCombinedCandlesticksChartsData` hook, the timestamps come back as strings in seconds

Just some initial findings as I'm poking around in here. Is updating the subgraphs to spit back data uniformly an option for us? Maybe not as part of this PR, but in the near future

File: sections/exchange/TradeCard/Charts/Types/AreaChart.tsx
Comment (kmeraz) [reply sections/exchange/TradeCard/Charts/Types/AreaChart.tsx:85]: Yeah, exactly. I'm rendering 1 x-axis as the day, and the second as the month.  Here's a screenshot of 2 months in the same view:
image

File: sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx
Comment (kmeraz) [reply sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx]: it turns out that the `useHistoricalRates` query does some parsing under the hood of the rates it fetches. 
- https://github.com/Synthetixio/js-monorepo/blob/%40synthetixio/data%402.46.5-ovm.1/packages/data/src/index.ts#L197
then:
- https://github.com/Synthetixio/js-monorepo/blob/ad1a23c6f03019cfa4e60049eb7a9aff87424dcf/packages/data/queries/rateUpdates/parse.ts#L18

so, for the purpose of this PR, I'm doing the same quick math on the timestamp we get from the `useCandlesticksQuery` as the one that we get from `useHistoricalRates` query, to keep things uniform. 

I'm going to follow up with a PR to have the Kwenta UI use the `useCandlesticksQuery` from the monorepo instead of from `kwenta`, and subsequently delete the query within Kwenta

File: sections/exchange/TradeCard/Charts/Types/AreaChart.tsx
Comment (kmeraz) [reply sections/exchange/TradeCard/Charts/Types/AreaChart.tsx:85]: Just need to offset that june label to the left

File: sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx
Comment (JChiaramonte7) [reply sections/exchange/TradeCard/Charts/Types/CandlesticksChart.tsx]: Sounds good. Thanks for digging into this

**Comment by JChiaramonte7** [comment]
image
Did you change this to month under every date to account for multiple months?

**Comment by kmeraz** [comment]
@JChiaramonte7 hey, yeah i could use another aesthetics opinion here. the unique months looked great, but they weren't aligning correctly. If you look at this screenshot, June should've been at the far left edge but instead the unique months are being centered along the x-axis.

image

So I pushed a commit with another approach where I'm stacking the date/month values instead, within the same XAxis component (so, only ever 1 XAxis component). We can set how often to render them (via the `interval` prop on the X-Axis). The ideal setup would be to use a media query to change the `interval` value based on the width of the screen. For example, when the screen is big, make `interval={0}` so no date/months are skipped. When the screen is tiny, maybe make it `interval={10}` so we can have 10 date/months skipped, so they don't overlap and collide with each other. The next step I was taking is to look at the best way to do media queries in javascript on our codebase, given that we use styled-components. Happy to hear another opinion from an aesthetics perspective if you have one.

**Comment by JChiaramonte7** [comment]
> @JChiaramonte7 hey, yeah i could use another aesthetics opinion here. the unique months looked great, but they weren't aligning correctly. If you look at this screenshot, June should've been at the far left edge but instead the unique months are being centered along the x-axis.
> 
> image
> 
> So I pushed a commit with another approach where I'm stacking the date/month values instead, within the same XAxis component (so, only ever 1 XAxis component). We can set how often to render them (via the `interval` prop on the X-Axis). The ideal setup would be to use a media query to change the `interval` value based on the width of the screen. For example, when the screen is big, make `interval={0}` so no date/months are skipped. When the screen is tiny, maybe make it `interval={10}` so we can have 10 date/months skipped, so they don't overlap and collide with each other. The next step I was taking is to look at the best way to do media queries in javascript on our codebase, given that we use styled-components. Happy to hear another opinion from an aesthetics perspective if you have one.

I think what you did looks better than unevenly spaced dates, but I would get an approval from Andrew because I know he didn't really want to stack the months vertically.

**Comment by JChiaramonte7** [comment]
https://recharts.org/en-US/api/XAxis#angle

Did you try adjusting the angle of the x-axis?

image

**Review by JChiaramonte7 — APPROVED** [review]

**Pull request merged**
